src/app/elevation/qgselevationprofilewidget.cpp

```diff
@@ -768,7 +768,7 @@ void QgsElevationProfileWidget::exportAsPdf()
   pageLayout.setMode( QPageLayout::FullPageMode );
   pdfWriter.setPageLayout( pageLayout );
   pdfWriter.setPageMargins( QMarginsF( 0, 0, 0, 0 ) );
-  pdfWriter.setResolution( 300 );
+  pdfWriter.setResolution( 1200 );
```

Comment:
(For the record, that's this value: https://github.com/qt/qtbase/blob/a984d98c971234c111dd144681402c1e20e9d806/src/gui/painting/qpdf.cpp#L1413-L1415)